system/modules/core/widgets/FileSelector.php

```diff
@@ -456,6 +456,12 @@ protected function convertValuesToPaths()
 			return;
 		}
 
+		// Do not convert values if path is not in the upload path
+		if ($this->path !== \Config::get('uploadPath'))
```

Comment:
Afaik the path does not store the value, but the eval option path. This is actually how it works in my environment as well.